diff --git a/node_modules/expo-sharing/android/src/main/java/expo/modules/sharing/SharingModule.kt b/node_modules/expo-sharing/android/src/main/java/expo/modules/sharing/SharingModule.kt
index 69ccba6..6dc6cc3 100644
--- a/node_modules/expo-sharing/android/src/main/java/expo/modules/sharing/SharingModule.kt
+++ b/node_modules/expo-sharing/android/src/main/java/expo/modules/sharing/SharingModule.kt
@@ -36,12 +36,15 @@ class SharingModule : Module() {
         val mimeType = params.mimeType
           ?: URLConnection.guessContentTypeFromName(fileToShare.name)
           ?: "*/*"
+        val sharingIntent = createSharingIntent(contentUri, mimeType)
         val intent = Intent.createChooser(
-          createSharingIntent(contentUri, mimeType),
+          sharingIntent,
           params.dialogTitle
-        )
+        ).apply {
+          addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+        }
         val resInfoList = context.packageManager.queryIntentActivities(
-          intent,
+          sharingIntent,
           PackageManager.MATCH_DEFAULT_ONLY
         )
         resInfoList.forEach {
@@ -116,6 +119,7 @@ class SharingModule : Module() {
   private fun createSharingIntent(uri: Uri, mimeType: String?) =
     Intent(Intent.ACTION_SEND).apply {
       putExtra(Intent.EXTRA_STREAM, uri)
+      clipData = android.content.ClipData.newRawUri(null, uri)
       setTypeAndNormalize(mimeType)
       addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
     }
